code/game/objects/items/weapons/storage/boxes.dm

```diff
@@ -338,6 +339,62 @@
 	new /obj/item/weapon/grenade/flashbang(src)
 	new /obj/item/weapon/grenade/flashbang(src)
 
+/obj/item/weapon/storage/box/firingpins
+	name = "box of firing pins"
+	desc = "A box of NT brand Firearm authentication devices - Needed to operate most weapons."
+
+/obj/item/weapon/storage/box/firingpins/fill()
+	..()
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	name = "box of firing pins"
+	desc = "A box of NT brand Firearm authentication devices - Needed to operate most weapons."
+
+/obj/item/weapon/storage/box/firingpins/fill()
+	..()
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+
+/obj/item/weapon/storage/box/testpins
+	name = "box of firing pins"
+	desc = "A box of NT brand Testing Authentication pins, allows guns to fire in designated areas."

```

Comment:
What designated areas? Explicitly saying that it only works in firing ranges here might be good.